engine/runtime/src/main/java/org/enso/interpreter/runtime/builtin/Builtins.java

```diff
@@ -340,29 +342,32 @@ private static Map<String, Method> readBuiltinMethodsMethods() {
     }
 
     return lines.stream()
-        .map(
-            line -> {
-              String[] builtinMeta = line.split(":");
-              if (builtinMeta.length != 2) {
-                throw new CompilerError("Invalid builtin metadata in: " + line);
-              }
-              String[] builtinName = builtinMeta[0].split("\\.");
-              if (builtinName.length != 2) {
-                throw new CompilerError("Invalid builtin metadata in : " + line);
-              }
-              try {
-                @SuppressWarnings("unchecked")
-                Class<BuiltinRootNode> clazz =
-                    (Class<BuiltinRootNode>) Class.forName(builtinMeta[1]);
-                Method meth = clazz.getMethod("makeFunction", Language.class);
-                // methodNodes.put(builtinMeta[0], meth);
-                return new AbstractMap.SimpleEntry<String, Method>(builtinMeta[0], meth);
-              } catch (ClassNotFoundException | NoSuchMethodException e) {
-                e.printStackTrace();
-                throw new CompilerError("Invalid builtin method " + line);
-              }
-            })
-        .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+            .map(
+                line -> {
+                  String[] builtinMeta = line.split(":");
+                  if (builtinMeta.length < 2 || builtinMeta.length > 4) {
+                    throw new CompilerError("Invalid builtin metadata in: " + line);
+                  }
```

Comment:
Again, it seems that with how the metadata is now generated we should be able to assume that `builtinMeta.length == 4` always. This would simplify the further code a little bit. Any reason why keep this length varying?